src/App.tsx

```diff
@@ -1,12 +1,58 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import React, { useEffect, useState } from 'react';
+import classNames from 'classnames';
+
+import { getTodos, USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+
+import { Filter } from './components/Filter';
+import { Notification } from './components/Notification';
+import { NewTodo } from './components/NewTodo';
+import { TodoList } from './components/TodoList';
+
+const FILTERS = {
+  all: 'all',
+  active: 'active',
+  completed: 'completed',
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState<boolean>(false);
+  const [error, setError] = useState<string | null>(null);
+  const [filter, setFilter] = useState<string>(FILTERS.all);
+  const [isTyping, setIsTyping] = useState<boolean>(false);
```

Comment:
there is no reason to specify simple types such as boolean or string
```suggestion
  const [isTyping, setIsTyping] = useState(false);
```